jupyter_core/paths.py

```diff
@@ -9,21 +9,78 @@
 
 
 import os
+import ast
 import sys
 import stat
 import errno
+import time
+import pathlib
 import tempfile
 import warnings
+import functools
+import traceback
+import importlib
 
 from contextlib import contextmanager
 
+import entrypoints
+
 pjoin = os.path.join
 
 # UF_HIDDEN is a stat flag not defined in the stat module.
 # It is used by BSD to indicate hidden files.
 UF_HIDDEN = getattr(stat, 'UF_HIDDEN', 32768)
 
 
+# The entry_point MUST be the name of a directory, consisting of only
+# letters, numbers, and underscores, adjacent to __init__.py
+JUPYTER_DATA_PATH_ENTRY_POINT = "jupyter_data_path"
```

Comment:
on closer reading of the spec, we should probably go with `jupyter_core.data_path` and `jupyter_core.config_path` (or drop the `path` bits), as this is a _new entry point group_.